kuronometer-core/src/main/scala/com/github/pedrovgs/kuronometer/Kuronometer.scala

```diff
@@ -0,0 +1,92 @@
+package com.github.pedrovgs.kuronometer
+
+import cats.free.Free
+import com.github.pedrovgs.kuronometer.KuronometerResults.{ConnectionError, KuronometerResult}
+import com.github.pedrovgs.kuronometer.app.KuronometerProgram
+import com.github.pedrovgs.kuronometer.free.algebra.{ReporterOps, ViewOps}
+import com.github.pedrovgs.kuronometer.free.domain.View.Message
+import com.github.pedrovgs.kuronometer.free.domain._
+import com.github.pedrovgs.kuronometer.free.interpreter.Interpreters
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.DurationFormatter.NanosecondsFormat.format
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.SummaryBuildStageExecutionFormatter
+
+class Kuronometer(implicit R: ReporterOps[KuronometerProgram], V: ViewOps[KuronometerProgram], interpreters: Interpreters) {
```

Comment:
Ideally you only impose these implicit constrains in those functions that need it and then this `class` would become an `object` with just a collection of functions where some have different constrains.